crates/polars-plan/src/dsl/function_expr/index_of.rs

```diff
@@ -27,11 +27,7 @@ pub(super) fn index_of(s: &mut [Column]) -> PolarsResult<Column> {
     let result = match is_sorted_flag {
         // If the Series is sorted, we can use an optimized binary search to
         // find the value.
-        IsSorted::Ascending | IsSorted::Descending
```

Comment:
`needle` being `null` still allows search sorted to be done in an optimized matter. Sorting flags guarantee that the nulls are either and the end or the beginning. So we need to check at most 1 element to know where all nulls are.